flink-cdc-common/src/main/java/org/apache/flink/cdc/common/event/DataChangeEvent.java

```diff
@@ -40,11 +40,22 @@ private DataChangeEvent(
             RecordData after,
             OperationType op,
             Map<String, String> meta) {
+        this(tableId, before, after, op, meta, null);
+    }
+
+    private DataChangeEvent(
+            TableId tableId,
+            RecordData before,
+            RecordData after,
+            OperationType op,
+            Map<String, String> meta,
```

Comment:
What about putting schema info in meta directly, and add one kind of MySqlReadableMetadata to extract it from SourceRecord.